2023-09-20-issue1-seed-identification-api-for-nachet-frontend.py

```diff
@@ -0,0 +1,151 @@
+        json_template (json): A JSON template for the OpenAI request.
+
+    This function performs the following steps:
+    1. Iterates through a list of seed values.
+    2. Checks if a JSON file for each seed exists and skips if it does.
+    3. Constructs an SQL query to retrieve data related to the seed from a database.
+    4. Sends the query to the database and fetches the retrieved data.
+    5. Concatenates the cleaned content into a single 'page.'
+    6. Sends a request to the Azure OpenAI endpoint to get a response.
+    7. Processes the response, extracting the name and saving it as a JSON file.
+    """
+    for url, seed_name in url_to_seed_mapping.items():
+        print("\nCurrent seed : " + seed_name)
+        seed_json_path = seed_name + ".json"
+
+        if nachet.json_file_exists(seed_data_path, seed_json_path):
+            print(f"JSON file {seed_json_path} exists in {seed_data_path}, skipping")
+        else:
+            query = """
+            SELECT
+                hc.md5hash,
+                REGEXP_REPLACE(hc.content, '<[^>]+>', '', 'g')
+            AS
+                cleaned_content
+            FROM
+                html_content hc
+            INNER JOIN
+                crawl c ON hc.md5hash = c.md5hash
+            WHERE
+                 c.url = '""" + url + """';
+            """
+            cursor.execute(query)
+            web_pages_fr_en = cursor.fetchall()
+
+            all_language_seed_page = ""
+            for row in web_pages_fr_en:
+                web_text = row.get('cleaned_content')
+                all_language_seed_page += web_text
+            page = all_language_seed_page
+            md5hash = row.get('md5hash') 
+
+            ### Get the images corresponding to the current page
+            query = """
+            SELECT DISTINCT
+                image_links[1] AS photo_link,
+                image_descriptions[1] AS photo_description
+            FROM (
+                SELECT
+                    (regexp_matches(content, 'src="([^"]+)"', 'g')) AS image_links,
+                    (regexp_matches(content, '<figcaption>(.*?)</figcaption>', 'gs'))
+                    AS image_descriptions
+                FROM html_content
+                WHERE md5hash = '""" + md5hash + """'
+            )
+            AS extracted_data;
+            """
+            cursor.execute(query)
+            images_fetch = cursor.fetchall()
+
+            image_information = ""
+            
+            for row in images_fetch:
+                image_links = row['photo_link']
+                image_descriptions = row['photo_description']
+                image_information += f"Image link: {image_links}"
+                image_information += f"\nImage description: {image_descriptions}\n\n"
+
+            print("Sending request for summary to Azure OpenAI endpoint...\n")
+            response = openai.get_chat_answer(system_prompt, user_prompt, json_template, page, image_information)  # noqa: E501
```

Comment:
Best fix the warning instead of suppressing it. If correctly configured, the Black extension fixes most of the lint issues on save (ctrl+s).